src/Utils/Strict.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+/**
+ * This file is part of the Nette Framework (https://nette.org)
+ * Copyright (c) 2004 David Grudl (https://davidgrudl.com)
+ */
+
+namespace Nette;
+
+use Nette\Utils\ObjectMixin;
+
+
+/**
+ * Strict class (access to undeclared members throws exceptions) with 'did you mean' hints for better experience.
+ */
+trait Strict
+{
+
+	/**
+	 * Call to undefined method.
+	 * @throws MemberAccessException
+	 */
+	public function __call($name, $args)
+	{
+		$class = method_exists($this, $name) ? 'parent' : get_class($this);
+		$items = (new \ReflectionClass($this))->getMethods(\ReflectionMethod::IS_PUBLIC);
```

Comment:
Well, that is my point. Should the static methods be suggested? Since they're static and there is no reason to call them as non-static.